app/models/ip_range.rb

```diff
@@ -0,0 +1,58 @@
+##
+# A class to model a set of IP ranges.  The default IPAddr library in ruby
+# can define broad IP ranges (e.g. 172.12.0.0/16 would define 172.12.*.*)
+# but doesn't handle custom ranges as well (e.g. 172.22.24.* - 172.22.255.*)
+# This class allows you to check if a given IP is in a network of IP ranges.
+#
+# There is a network of IP ranges injected by default, but an object can be
+# injected that provides an array of singleton IP addresses (or ranges defined
+# by the slash prefixes) and an mutli-dimensional array of custom IP ranges.
+# Example network definition:
+# example_network = OpenStruct.new(
+#   singletons: ['172.12.0.0/16'],
+#   ranges: [['172.12.12.0', '172.12.56.255']]
+# )
+#
+# Example Usage:
+# IPRange.includes?('172.12.52.21')
+# => false
+# IPRange.new('172.12.52.21', example_network).includes?
+# => true
+class IPRange
+  attr_reader :ip_to_check, :network
+
+  def initialize(ip, network_definition = self.class.default_network)
+    @ip_to_check = IPAddr.new(ip)
+    @network = network_definition
+  end
+
+  def included?
+    singletons_include_ip? || ranges_include_ip?
+  end
+
+  class << self
+    def includes?(ip_to_check)
+      new(ip_to_check).included?
+    end
+
+    def default_network
+      Settings.stanford_ips
+    end
+  end
+
```

Comment:
They are two completely unrelated methods.  They are both providing class methods so we can say things like `self.class.default_network` as is in the initializer or `IPRange.includes?` as is used in the helper.